reference/sql/characterset-and-collation.md

```diff
@@ -0,0 +1,374 @@
+
+    The `SET NAMES 'charset_name'` statement is equivalent to the following statement combination:
+
+    ```sql
+    SET character_set_client = charset_name;
+    SET character_set_results = charset_name;
+    SET character_set_connection = charset_name;
+    ```
+
+    `COLLATE` is optional, if absent, the default collation of the `charset_name` is used.
+
++ `SET CHARACTER SET 'charset_name'`
+
+    Similar to `SET NAMES`, the `SET NAMES 'charset_name'` statement is equivalent to the following statement combination:
+
+    ```sql
+    SET character_set_client = charset_name;
+    SET character_set_results = charset_name;
+    SET collation_connection = @@collation_database;
+    ```
+
+## Optimization levels of character sets and collations
+
+String > Column > Table > Database > Server > Cluster
+
+## General rules on selecting character sets and collation
+
++ Rule 1: If you specify `CHARACTER SET charset_name` and `COLLATE collation_name`, then `CHARACTER SET charset_name` and `COLLATE collation_name` are used directly.
++ Rule 2: If you specify `CHARACTER SET charset_name` and do not specify `COLLATE collation_name`, then `CHARACTER SET charset_name` and the default comparison collation of `CHARACTER SET charset_name` are used.
++ Rule 3: If you specify neither `CHARACTER SET charset_name` nor `COLLATE collation_name`, the character set and collation with higher optimization levels are used.
+
+## Validity check of characters
+
+For the specified `utf8` or `utf8mb4` character set, TiDB only supports the valid `utf8` character, and reports the `incorrect utf8 value` error when the character is invalid. This validity check of characters in TiDB is compatible with MySQL 8.0 but incompatible with MySQL 5.7 or earlier versions.
+
+To disable this error reporting, use `set @@tidb_skip_utf8_check=1;` to skip the character check.
+
+For more information, see [Connection Character Sets and Collations in MySQL](https://dev.mysql.com/doc/refman/5.7/en/charset-connection.html).
+
+## Collation support framework
+
+The syntax support and semantic support for the collation are influenced by the [`new_collations_enabled_on_first_bootstrap`](/reference/configuration/tidb-server/configuration-file.md#new_collations_enabled_on_first_bootstrap) configuration item. The syntax support and semantic support are different. The former indicates that TiDB can parse and set collations. The latter indicates that TiDB can correctly use collations when comparing strings.
+
+Before v4.0, TiDB only supports syntactically parsing most of the MySQL collations but semantically takes all collations as binary collations, which is the [old framework for collations](#old-framework-for-collations).
+
+Since v4.0, TiDB supports semantically parsing different collations and strictly following the collations when comparing strings, which is the [new framework for collations](#new-framework-for-collations).
```

Comment:
```suggestion
Since v4.0, TiDB supports a [new framework for collations](#new-framework-for-collations). In this framework, TiDB semantically parses different collations and strictly follows the collations when comparing strings.
```